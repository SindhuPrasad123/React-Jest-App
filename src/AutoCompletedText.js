import React from 'react';
import countries from './Countries';
import './App.css';

export default class AutoCompletedText extends React.Component{

    constructor(props){
        super(props)
        this.state = {
            suggestions: [],
            text: ''
        }
    }

    onTextChange = (e) => {
        const value = e.target.value;
        let suggestions = [];
        if(value.length > 0){
            const regex = new RegExp(`^${value}`, 'i');
            suggestions = countries.sort().filter(v => regex.test(v))
        }

        this.setState(() => ({
            suggestions,
            text: value
        }))
    }

    selectedText(value) {
        console.log("Inside selected Text");
        this.setState(() => ({
            text: value,
            suggestions: [],
        }))
    }

    renderSuggestions = () => {
        let { suggestions } = this.state;
        if(suggestions.length === 0){
            return null;
        }
        return (
           
            <ul id= "list1">
                {
                    suggestions.map((item, index) => (<li id = "list2" key={index} onClick={() => this.selectedText(item)}>{item}</li>))
                }
            </ul>
        );
    }
    
    render() {
        const { text, suggestions } = this.state;
        return(
            <div id="notebooks">
                <h2>Auto Completed</h2>
                <input id="query" type="text" onChange={this.onTextChange} value={text}/>
                {this.renderSuggestions()}
                <span>Suggestions: {suggestions.length}</span>
            </div>
        );
    }

}