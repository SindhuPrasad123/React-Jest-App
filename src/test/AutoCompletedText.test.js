import React from 'react'
import renderer from 'react-test-renderer';
import AutoCompletedText from '../AutoCompletedText'
import { shallow, mount } from 'enzyme';
import {sinon} from 'jest-sinon'

describe('<AutoCompleteText /', () =>{


    it('Auto Text Correctly', () => {
        const tree = renderer.create(<AutoCompletedText />)
        expect(tree).toMatchSnapshot();
    })


    it('AutoCompletedText on text change', () => {
        const wrapper = shallow(<AutoCompletedText/>);
        wrapper.find('input').simulate('change', {target: {value: 'Jap'}});
        const instance = wrapper.instance();
        expect(instance.state.suggestions).toEqual(['Japan']);


    })

    it('AutoCompleted on text not selected', () => {
       const wrapper = shallow(<AutoCompletedText/>);
       const instance = wrapper.instance();
       expect(instance.state.text).toEqual('');
    })

    it('AutoCompleted on text selected', () => {
        const wrapper = mount(<AutoCompletedText value = {'Jap'}/>);
        wrapper.setState({suggestions:['Jamaica', 'Japan'], text:''})
       wrapper.find('li').first().simulate('click');
       const instance = wrapper.instance();
       expect(instance.state.text).toEqual('Jamaica');
     })

})



