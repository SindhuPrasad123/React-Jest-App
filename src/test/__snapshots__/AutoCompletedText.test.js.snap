// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoCompleteText / Auto Text Correctly 1`] = `
<div
  id="notebooks"
>
  <h2>
    Auto Completed
  </h2>
  <input
    id="query"
    onChange={[Function]}
    type="text"
    value=""
  />
  <span>
    Suggestions: 
    0
  </span>
</div>
`;

exports[`Auto Text Correctly 1`] = `
<div
  id="notebooks"
>
  <h2>
    Auto Completed
  </h2>
  <input
    id="query"
    onChange={[Function]}
    type="text"
    value=""
  />
  <span>
    Suggestions: 
    0
  </span>
</div>
`;
